gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageGrpcIntegrationTest.java

```diff
@@ -129,13 +132,18 @@ public void testOpen() throws IOException {
 
   @Test
   public void testOpenWithTDEnabled() throws IOException {
-    GoogleCloudStorageOptions storageOptions =
+    String grpcServerAddress = System.getenv("GCS_TEST_GRPC_SERVER_ADDRESS");
+    logger.atInfo().log("Printing env var for grpcServerAddress: " + grpcServerAddress);
+
+    GoogleCloudStorageOptions.Builder optionsBuilder =
         GoogleCloudStorageTestHelper.getStandardOptionBuilder()
             .setGrpcEnabled(true)
-            .setTrafficDirectorEnabled(true)
-            .build();
+            .setTrafficDirectorEnabled(true);
+    if (grpcServerAddress != null) optionsBuilder.setGrpcServerAddress(grpcServerAddress);
```

Comment:
This isn't ideal. We seem to test two different scenarios in one test.

we should relook at the structure of the test

Reply:
I have restructured it now, so each test would run twice with TD enabled/disabled. Let me know if you have any questions and we can discuss offline.